src/x86/bmi.rs

```diff
@@ -0,0 +1,266 @@
+//! Bit Manipulation Instruction (BMI) Set 1.0.
+//!
+//! The reference is [Intel 64 and IA-32 Architectures Software Developer's
+//! Manual Volume 2: Instruction Set Reference,
+//! A-Z](http://www.intel.de/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-instruction-set-reference-manual-325383.pdf).
+//!
+//! [Wikipedia](https://en.wikipedia.org/wiki/Bit_Manipulation_Instruction_Sets#BMI1_.28Bit_Manipulation_Instruction_Set_1.29)
+//! provides a quick overview of the available instructions.
+
+#[allow(dead_code)]
+extern "platform-intrinsic" {
```

Comment:
Yeah so I'm explicitly trying not to use `platform-intrinsic` since I'd like to do away with that machinery. If you take a look at the other files, we link with the llvm intrinsics directly. My guess (based on [llvm intrinsic defs](https://gist.githubusercontent.com/anonymous/a25d3e3b4c14ee68d63bd1dcb0e1223c/raw/fb798be53a7b29f9b28a0c7a9e2b9a16ae915f71/allintrinsics) is that you probably want

```rust
extern {
    #[link_name = "llvm.x86.bmi.bextr.32"]
    fn bextr_u32(x: u32, y: u32) -> u32;
    #[link_name = "llvm.x86.bmi.bextr.64"]
    fn bextr_u32(x: u64, y: u64) -> u64;
}
```

Reply:
@BurntSushi I did not knew this, should be fixed now.

Reply:
Yeah technically we're still using `platform-intrinsic` for the cross platform simd llvm intrinsics (see `src/simd_llvm.rs`), but I'm trying to avoid them other than that. In particular, the platform intrinsics, IIRC, can't actually represent all of the vendor intrinsics. I think it's simpler to just write them out explicitly in Rust code instead of describing them in JSON.